Minor polishing suggested by warnings from clang

gc00 opened the pull request:
All of these changes were suggested by warnings when compiling
with the clang-3.4 compiler.  Most are self-explanatory.

Here are reasons for two of the more obscure changes:

1. shareddata.cpp :  size_t -> int
    size_t is an unsigned data type.  So, the i>=0 condition is always true.
    This is a bug, since we want to allow i-- to cause i to become negative, and return false in that case.

2. procselfmaps.cpp
    ProcSelfMaps::ProcSelfMaps() initializes the private field members in a certain order.  But in include/procselfmaps.h, those fields were declared in a different order.
    Clang warns about this.  Now the initialization respects the order in which the fields were declared.

rohgarg commented on src/mtcp/mtcp_check_vdso.ic:
```diff
@@ -335,11 +335,12 @@ void mtcp_check_vdso(char **environ)
          */
 #define ARCH_STACK_DEFAULT_SIZE (256 * 1024 * 1024)
 
+        int rlimit_failed = 0;
```
nit: Can you please fix the indentation on this?

gc00 commented:
@rohgarg, I fixed the indentation.  mtcp_check_vdso.ic was inconsistent about whether to use TAB or 8&nbsp;spaces before.  I added a commit to use 8&nbsp;spaces always.

rohgarg commented:
Looks good to me.
